orbstation/xcard/xenotoxin.dm

```diff
@@ -0,0 +1,113 @@
+#define DAMAGE_BURST_RATE 5
+#define JITTER_TIME 10 SECONDS
+// note that unlike tox damage, stamina damage, etc, organ damage is NOT multiplied by DAMAGE_BURST_MUL or increased by num_damage_bursts
+#define ORGAN_DAMAGE_MIN 3
+#define ORGAN_DAMAGE_MAX 10
+
+// initial chance of a damage burst being "bad" (dealing extra damage, causing jitters, and displaying a scarier warning message)
+#define BAD_BURST_CHANCE_MIN 10
+// the health value below which having lower health starts to increase the chance of having a bad damage burst
+#define BAD_BURST_HEALTH_RAMP_THRESHOLD 50
+// the percentage above which having low health no longer increases the probability of a bad damage burst
+#define BAD_BURST_CHANCE_SOFTCAP 60
+// chance to scream on a bad damage burst
+#define SCREAM_PROB 50
+
+/datum/reagent/toxin/xenotoxin
+    name = "Xenotoxin"
+    description = "A powerful venom injected by immature xenomorphs when they fail infect a host. \
+                    Lingers in the system for a long time, causing the victim to continuously weaken."
+    color = "#1a011c" // not sure the best color, so i went with xeno-ish purple/black
+    toxpwr = 0.25 // deals low damage but spends a long, long time in your system
+    ph = 1
+    taste_description = "acrid bile"
+    penetrates_skin = TOUCH
+    metabolization_rate = 0.1 * REAGENTS_METABOLISM
+
+    // number of times a damage burst has been triggered
+    // the more bursts you suffer, the more damage each one does and the higher the chance that you'll get "bad bursts" that deal extra damage
+    var/num_damage_bursts = 0
+
+/datum/reagent/toxin/xenotoxin/on_mob_life(mob/living/carbon/victim, delta_time, times_fired)
+
+    ..()
+
+    // apply a fluctuating amount of stamina damage in little chunks over time
+    if (DT_PROB(STAM_DAMAGE_RATE, delta_time))
+        victim.adjustStaminaLoss(rand(STAM_DAMAGE_MIN, STAM_DAMAGE_MAX) * REM * delta_time, 0)
+
+    // check to see if we should apply a damage burst
+    // these deal sudden, larger amounts of toxin and stamina damage, and display a warning message
+    if (DT_PROB(DAMAGE_BURST_RATE, delta_time))
+
+        // base chance of a bad burst, which increases as victim's health decreases
+        var/bad_burst_base_chance = BAD_BURST_HEALTH_RAMP_THRESHOLD - victim.health
+
+        // actual chance, which is clamped to a min/max and increases with the number of previous bursts
+        // note that health decreasing below a certain point will no longer increase the chance, but further bursts will
+        // always make bad ones more likely
+        var/bad_burst_chance = clamp(bad_burst_base_chance, BAD_BURST_CHANCE_MIN, BAD_BURST_CHANCE_SOFTCAP) + num_damage_bursts
+
+        if (prob(bad_burst_chance))
+            to_chat(victim, span_danger("Your heartrate spikes, and it feels like acid is running through your veins!"))
+            damage_burst_bad(victim, DAMAGE_BURST_MUL + num_damage_bursts, delta_time)
+
```

Comment:
Prefer if you don't place a newline before `else` either

Reply:
Keeping for now, for personal style reasons.

Reply:
Hm, on second thought, maybe you're right.